h2/src/main/org/h2/engine/Session.java

```diff
@@ -226,10 +229,25 @@ public SubQueryInfo getSubQueryInfo() {
         return subQueryInfo;
     }
 
-    public void setParsingView(boolean parsingView) {
+    public void setParsingView(boolean parsingView, String viewName) {
```

Comment:
I added this so persistent ctes could know who created them - and somehow record that they are joined to the parent view lifecycle. (That lifecycle bit is not implemented yet, especially for the unused ctes.)